packages/ui/src/lib/CommitLines/AvatarImage.svelte

```diff
@@ -0,0 +1,59 @@
+<script lang="ts">
+	import { tooltip } from '$lib/utils/tooltip';
+
+	interface Props {
+		srcUrl: string;
+		tooltipText: string;
+		altText: string;
+	}
+
+	const { srcUrl, tooltipText, altText, ...rest }: Props = $props();
+
+	let isLoaded = $state(false);
+</script>
+
+<div class="image-wrapper">
+	<img
+		class="avatar"
+		alt={altText}
+		src={srcUrl}
+		onload={() => (isLoaded = true)}
+		width="100"
+		height="100"
+		class:hidden={!isLoaded}
+		use:tooltip={tooltipText}
+		{...rest}
+	/>
```

Comment:
should we have `loading="lazy"` to this?

Reply:
Yeah we can definitely try it, this seems like an appropriate place to lazy load the avatar images.

I think there's a bigger issue though when loading large lists of commits though, it seems like the entire commit card is a bit janky to load when being rendered for the first time. Not just this small image part. I've been doing some more digging, but haven't found anything concrete yet.